src/ui/SmartSnippet/SmartSnippet.ts

```diff
@@ -301,19 +301,21 @@ export class SmartSnippet extends Component {
   }
 
   private renderSourceTitle() {
-    return this.buildLink(Utils.getFieldValue(this.lastRenderedResult, <string>this.options.titleField), SOURCE_TITLE_CLASSNAME);
+    const link = this.buildLink();
+    link.innerText = Utils.getFieldValue(this.lastRenderedResult, <string>this.options.titleField);
+    link.classList.add(SOURCE_TITLE_CLASSNAME);
```

Comment:
This should be part of `buildLink` and should use the functionalities of `DomUtils` ($$.text(), $$.class) etc.